### PR TITLE
scheduler: add test for creating evict-leader-scheduler twice

### PR DESCRIPTION
**Pull request by okJiang**
### What problem does this PR solve?

Issue Number: Close https://github.com/tikv/pd/issues/8756

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

**Comment by okJiang** [comment]
Grant-leader exists bug. It will not work in some specific condition... I have to find it firstly. So this pr hold

**Comment by okJiang** [comment]
> Grant-leader exists bug. It will not work in some specific conditions... I have to find it firstly. So this pr hold

The issue has been fixed by https://github.com/tikv/pd/pull/8623/files#diff-a220672e83f3e5fa65303ae268e69c0e4bcabebd6d3b5d1c81898504f9ac5095R496-R503.

Assuming there are three nodes: store0, store1, and store2, we set up `scheduler add grant-leader-scheduler store0`.

Then, when the balance-leader-scheduler selects a target, it cannot be store0, but it may choose store1 or store2. Let's assume target=store1. Ultimately, when creating the transfer-leader-operator for the target, there will be no filtering on the source, which leads to the creation of a transfer-leader-operator from store0 to store1.

This is a grant-leader-scheduler problem. So we do not need to pick it to previous versions.

**Review by lhy1024 — APPROVED** [review]

File: tests/integrations/realcluster/scheduler_test.go [hunk 1]
```diff
@@ -201,3 +203,72 @@ func (s *schedulerSuite) TestRegionLabelDenyScheduler() {
...
+
+	var i int
+	evictLeader := func() {
+		re.NoError(pdHTTPCli.CreateScheduler(ctx, types.EvictLeaderScheduler.String(), uint64(region1.Leader.StoreID)))
+		// if the second evict leader scheduler cause the pause-leader-filter
+		// disable, the balance-leader-scheduler need some time to transfer
+		// leader. See details in https://github.com/tikv/pd/issues/8756.
+		if i == 1 {
+			time.Sleep(3 * time.Second)
+		}
```
Comment (JmPotato) [inline tests/integrations/realcluster/scheduler_test.go:226]: I think we should cover this kind of waiting inside the `testutil.Eventually` below.

**Review by JmPotato — APPROVED** [review]

**Pull request merged**